src/aggregator/client/queue.go

```diff
@@ -131,92 +139,108 @@ func newInstanceQueue(instance placement.Instance, opts Options) instanceQueue {
 		connOpts           = opts.ConnectionOptions().
 					SetInstrumentOptions(connInstrumentOpts).
 					SetRWOptions(opts.RWOptions())
-		conn          = newConnection(instance.Endpoint(), connOpts)
-		iOpts         = opts.InstrumentOptions()
-		queueSize     = opts.InstanceQueueSize()
-		maxBatchSize  = opts.MaxBatchSize()
-		writeInterval = opts.BatchFlushDeadline()
+		conn      = newConnection(instance.Endpoint(), connOpts)
+		iOpts     = opts.InstrumentOptions()
+		queueSize = opts.InstanceQueueSize()
 	)
+
+	// Round up queue size to power of 2.
```

Comment:
The code seems to be rounding down the number rather than rounding up :)

Reply:
This will round up as expected.

Reply:
done, it def rounds up :)